pkg/build/strategies/sti/sti.go

```diff
@@ -306,9 +306,13 @@ func (builder *STI) PostExecute(containerID, location string) error {
 }
 
 func (builder *STI) getPreviousImage() string {
+	if !builder.incremental || !builder.config.RemovePreviousImage {
+		return ""
+	}
```

Comment:
seems like this if-logic does belong in postexecute, because what postexecute does is based on whether this is an incremental build or not.  hiding that logic in two different methods that have to stay in sync seems like a bad idea.  (getPreviousImage and removePreviousImage are now tightly coupled in a way that might not be obvious to someone editing one or the other of them).